lib/stack_master/aws_driver/s3.rb

```diff
@@ -0,0 +1,70 @@
+require 'digest/md5'
+
+module StackMaster
+  module AwsDriver
+    class S3ConfigurationError < StandardError; end
+
+    class S3
+      attr_accessor :region, :bucket, :prefix, :files
+
+      def initialize(region: nil, bucket: nil, prefix: nil, files: nil)
```

Comment:
The driver classes in stack master are designed to be analogous to the AWS SDK's Client classes. I think only region should be passed in as the initializer. Otherwise, this is more of a single use UploadFiles command class.  One of the main reasons we have these drivers is to enable us to easily run integration tests with cucumber & aruba without hitting AWS at all by using a fake/test driver (see the CloudFormation test driver for an example).

As for how to test, I'm not sure I understand your question properly. Are you asking how to test the initializer? If so I would say that it is unnecessary, the assignments will be tested indirectly through the upload files method, the only thing that might warrant a test is validating the bucket config.